Woof/Woof/View/SitterFlow/SitterOnboardingView.swift

```diff
@@ -0,0 +1,85 @@
+import SwiftUI
+
+/// A view to onboard the sitter by filling in the mandatory information about themselves.
+struct SitterOnboardingView: View {
+    /// View model responsible to manage data from model layer
+    @StateObject var viewModel = SitterProfileViewModel()
+
+    var body: some View {
+        VStack {
+            Text(welcomeText)
+                .foregroundColor(Color.App.purpleDark)
+                .font(.title)
+                .padding(.bottom, AppStyle.UIElementConstant.minPadding)
+
+            Text(onboardingText)
+                .foregroundColor(Color.App.purpleDark)
+
+            EditSitterInformationView(
+                name: $viewModel.name,
+                surname: $viewModel.surname,
+                phone: $viewModel.phone,
+                city: $viewModel.city,
+                bio: $viewModel.bio,
+                pricePerHour: $viewModel.pricePerHour
+            )
+
+            if viewModel.mandatoryFieldsAreEmpty {
+                Text(mandatoryPlaceholderText)
+                    .foregroundColor(Color.App.purpleDark)
+                    .padding()
+                    .font(.system(.footnote))
+            }
+
+            Button(proceedButtonTitle) {
+                Task {
+                    await viewModel.save()
+                }
+            }
+            .buttonStyle(CapsuleWithWhiteText())
+            .padding()
+            .disabled(viewModel.mandatoryFieldsAreEmpty)
+        }
+        .padding()
+        .overlay(
+            Group {
+                if viewModel.isSavingData {
+                    Color.white.opacity(AppStyle.UIElementConstant.opacityLevelForProgressViewBackground)
+                    ProgressView()
+                        .progressViewStyle(CircularProgressViewStyle())
```

Comment:
Does it make sense to extract this into a separate view? For now, we use it in the onboarding view and profile view. But I'm sure we will reuse it for owner flow as well.

Reply:
Created `CustomProgressView` with 4488b64

Reply:
I meant to  extract a white transparent background and progress view, it doesn't make sense to extract only the progress view

Reply:
Could you replace it with the new view in the changed code in `SitterProfileView`as well?

Reply:
Yes, updated with 2a4848e